std/node/_fs_dir.ts

```diff
@@ -0,0 +1,111 @@
+import Dirent from "./_fs_dirent.ts";
+
+export default class Dir {
+  private dirPath: string | Uint8Array;
+  private files: Dirent[] = [];
+  private filesReadComplete = false;
+
+  constructor(path: string | Uint8Array) {
+    this.dirPath = path;
+  }
+
+  get path(): string {
+    if (this.dirPath instanceof Uint8Array) {
+      return new TextDecoder().decode(this.dirPath);
+    }
+    return this.dirPath;
+  }
+
+  /**
+   * NOTE: Deno doesn't provide an interface to the filesystem like readdir
+   * where each call to readdir returns the next file.  This function simulates this
+   * behaviour by fetching all the entries on the first call, putting them on a stack
+   * and then poping them off the stack one at a time.
```

Comment:
We should support this. Please add a link to this issue I've created https://github.com/denoland/deno/issues/4218

Reply:
link now added